src/NUnitFramework/framework/Internal/TestNameGenerator.cs

```diff
@@ -444,65 +473,71 @@ public FixedTextFragment(string text)
                 _text = text;
             }
 
-            public override string GetText(MethodInfo method, object[] args)
+            public override string GetText(Test test, object[] args, bool argsAreDisplayNames)
             {
                 return _text;
             }
         }
 
         private class MethodNameFragment : NameFragment
         {
-            public override string GetText(MethodInfo method, object[] args)
+            public override string GetText(Test test, object[] args, bool argsAreDisplayNames)
             {
+                var method = test.Method;
+                if (method == null) return null;
+
                 var sb = new StringBuilder();
 
                 sb.Append(method.Name);
 
                 if (method.IsGenericMethod)
-                    AppendGenericTypeNames(sb, method);
+                    AppendGenericTypeNames(sb, method.MethodInfo);
```

Comment:
Why does this need to do `method.MethodInfo` when none of the other uses of `method` have changed?

Reply:
This code used to be inside the removed overload with parameter `MethodInfo method`. Now it's inside a `Test` overload and I did `var method = test.Method` rather than `var method = test.Method.MethodInfo`. Now most of the code is operating on our IMethodInfo rather than SR.MethodInfo which I think is what we're meant to do with `IMethodInfo` when possible.